internal/core/command/types.go

```diff
@@ -0,0 +1,63 @@
+/*******************************************************************************
+ * Copyright 2019 Dell Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ *******************************************************************************/
+package command
+
+import (
+	"bytes"
+	"github.com/edgexfoundry/edgex-go/internal"
+	"github.com/edgexfoundry/go-mod-core-contracts/models"
+	"net/http"
+)
+
+// DefaultErrorCode the default value used when an error has occurred and no response code was received.
+const DefaultErrorCode = 0
+
+// ErrDeviceLocked error indicating the locked state of a device
+type ErrDeviceLocked struct{}
+
+func (ErrDeviceLocked) Error() string {
+	return "The device is in admin locked state"
+}
+
+// ServiceCommand type which encapsulates command information to be sent to the command service.
+type ServiceCommand struct {
```

Comment:
Recommend `serviceCommand` so that the type is not exported. In general, I find it enforces utilization of an interface if the implementing types are not exportable. Granted that has a more firm boundary in a more complex package structure, but good to get in the habit.